lib/rules/no-mixed-operators.js

```diff
@@ -167,13 +180,13 @@ module.exports = {
          */
         function reportBothOperators(node) {
             const parent = node.parent;
-            const left = (parent.left === node) ? node : parent;
-            const right = (parent.left !== node) ? node : parent;
+            const left = ((parent.left || parent.test) === node) ? node : parent;
+            const right = ((parent.left || parent.test) !== node) ? node : parent;
             const message =
                 "Unexpected mix of '{{leftOperator}}' and '{{rightOperator}}'.";
             const data = {
-                leftOperator: left.operator,
-                rightOperator: right.operator
+                leftOperator: left.operator || "ternary operator",
```

Comment:
I'm slightly concerned about this hardcoded string, as it will make future internationalization/localization of this message more difficult. I would prefer that there be another message entirely, or else maybe use a string like `?:`.